### ISSUE TITLE
[FP] Substring comparison flaws

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1149]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                         |
| Plugin         |  Php Inspections (EA Ultimate), 2.0.11 |
| Language level | PHP 7.3                 |

Current behaviour (description/screenshot:)
--
The following code launch a "Substring comparison flaws" but it's working.
```php
$str2 = "\r";
if (substr($str2, -1) === "\r") {
    echo "true";
}
```
[3v4l link](https://3v4l.org/GNcCM)

Expected behaviour
--
The warning should not appear because the code is working as expected

Interestingly, if we replace \r by \t or \n the warning disappears. It seems to be specific to \r.

**Comment by kalessil** [comment]
Ouch, good finding @orklah. Fixed!

**Comment by orklah** [comment]
I have another strange case with this inspection:

https://3v4l.org/UJti0

This launch the warning and propose to replace 4 to 0. If we do it, the code who was working doesn't work anymore

**Comment by kalessil** [comment]
Ok, fixed this case as well.
